CMakeLists.txt

```diff
@@ -1,58 +1,195 @@
-   Secure Channel, \"OPENSSL\", \"LIBRESSL\", \"AUTO\",\ or \"OFF\". These options are
-   case-sensitive. The default is \"AUTO\". Note\ that SCRAM-SHA-1 is
-   required for authenticating to MongoDB 3.0 and later.")
-
-set_property(CACHE ENABLE_SSL PROPERTY STRINGS "DARWIN;WINDOWS;OPENSSL;LIBRESSL;AUTO;OFF")
-
-set (ENABLE_SASL AUTO CACHE STRING
-   "Enable SASL authentication (Kerberos). Options are \"CYRUS\" to use Cyrus
-   SASL, \"SSPI\" to use Windows Native SSPI, \"AUTO\",\ or \"OFF\". These options are case-sensitive.")
-
-set_property(CACHE ENABLE_SASL PROPERTY STRINGS "CYRUS|SSPI|AUTO|OFF")
-
-set (ENABLE_STATIC AUTO CACHE STRING "Build and install static libbson/libmongoc. Set to ON/AUTO/OFF/BUILD_ONLY/DONT_INSTALL, default AUTO.")
-option (ENABLE_TESTS "Build MongoDB C Driver tests." ON)
-option (ENABLE_EXAMPLES "Build MongoDB C Driver examples." ON)
-set (ENABLE_SRV AUTO CACHE STRING "Support mongodb+srv URIs. Set to ON/AUTO/OFF, default AUTO.")
-option (ENABLE_MAINTAINER_FLAGS "Use strict compiler checks" OFF)
-option (ENABLE_AUTOMATIC_INIT_AND_CLEANUP "Enable automatic init and cleanup (GCC only)" ON)
-option (ENABLE_CRYPTO_SYSTEM_PROFILE "Use system crypto profile (OpenSSL only)" OFF)
-option (ENABLE_TRACING "Turn on verbose debug output" OFF)
-option (ENABLE_COVERAGE "Turn on compile options for lcov" OFF)
-set (ENABLE_SHM_COUNTERS AUTO CACHE STRING "Enable memory performance counters that use shared memory on Linux. Set to ON/AUTO/OFF, default AUTO.")
-set (ENABLE_MONGOC ON CACHE STRING "Whether to build libmongoc. Set to ON/OFF, default ON.")
-set (ENABLE_BSON AUTO CACHE STRING "Whether to build libbson. Set to ON/AUTO/SYSTEM, default AUTO.")
-set (ENABLE_SNAPPY AUTO CACHE STRING "Enable snappy support. Set to ON/AUTO/OFF, default AUTO.")
-set (ENABLE_ZLIB AUTO CACHE STRING "Enable zlib support")
-set (ENABLE_ZSTD AUTO CACHE STRING "Enable zstd support. Set to ON/AUTO/OFF, default AUTO.")
-option (ENABLE_MAN_PAGES "Build MongoDB C Driver manual pages." OFF)
-option (ENABLE_HTML_DOCS "Build MongoDB C Driver HTML documentation." OFF)
-option (ENABLE_EXTRA_ALIGNMENT
-   "Turn on extra alignment of libbson types. Set to ON/OFF, default ON.\
- Required for the 1.0 ABI but better disabled."
-   ON
+cmake_minimum_required (VERSION 3.15)
+
+project (mongo-c-driver C)
+
+list (APPEND CMAKE_MODULE_PATH
+      "${PROJECT_SOURCE_DIR}/build/cmake"
+      "${PROJECT_SOURCE_DIR}/build/cmake/make_dist"
+      )
+
+include (MongoSettings)
+
+# Subcomponents:
+mongo_bool_setting(ENABLE_MONGOC "Enable the build of the mongo-c-driver libraries")
+mongo_bool_setting(
+   USE_SYSTEM_LIBBSON [[Do not build libbson, and instead use an existing one via find_package()]]
+   DEFAULT VALUE OFF
+   ADVANCED
+)
+
+# Deprecated option "ENABLE_BSON" is gone, but old caches may rely on SYSTEM to
+# disable the libbson build. Allow ENABLE_BSON=SYSTEM only if USE_SYSTEM_LIBBSON
+# is also true, to allow both ENABLE_BSON=SYSTEM and USE_SYSTEM_LIBBSON=TRUE to
```

Comment:
Allowing both options seems helpful. Consider adding a bullet to the unreleased section of the NEWS to warn users of the change:

> The `ENABLE_BSON` configuration option has been removed. Downstream build scripts using `ENABLE_BSON=SYSTEM` should be updated to use `USE_SYSTEM_LIBBSON=ON`. If the build script needs to support multiple versions of the C driver, add both `ENABLE_BSON=SYSTEM` and `USE_SYSTEM_LIBBSON=TRUE`.